MUlti-sample SIgle Cell deconvolution (MuSiC)
=============================================

`MuSiC` is a bioinformatics analysis pipeline that leverages single-cell sequencing data to estimate cell type proportions of RNA-seq data from bulk tissue samples.

![MuSiC\_pipeline](image/pipeline.png)

How to cite `MuSiC`
-------------------

When using this work in any way, please cite the following publication:

Installation
------------

``` r
# install devtools if necessary
install.packages('devtools')

# install the wdec package
devtools::install_github('xuranw/MuSiC')

# load
library(MuSiC)
```

Examples
--------

We reproduce here the analysis detailed in ... 

-   Total gene expression data were obtained from GEO dataset [GSE50244](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50244), which contains read counts data on gobal transcriptomic analysis of bulk human pancreatic islets to study glucose metabolism in healthy and hyper-glycemic conditions (Fadista et al. 2014).

-   Single cell RNA-seq data were generated by ArrayExpress dataset [E-MTAB-5060](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-5061/), which contains read counts data on single cell mRNA-seq of human pancreatic islets from 10 donors, 6 healthy donors and 4 Type 2 diabetes donors (Segerstolpe et al. 2016). We used single cell data from 6 healthy donors as reference in deconvolution.

### Data Preparation

Bioconduct base package provides `ExpressionSet` class, which is a convenient data structure to hold expression data along with sample/feature annotation. Here we use two `ExpressionSet` objects to handle the bulk and single cell data respectively.

#### Bulk data

The dataset's [GEO entry (GSE50244)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50244) contains raw RNA-seq and sample annotation data. For the purpose of this vignette, we will use the read counts data on the [data download page]().

``` r
# Download GEO dataset from Github
GSE50244.bulk.eset = readRDS('https://github.com/xuranw/MuSiC/tree/master/vignettes/data/GSE50244bulkeset.rds')

GSE50244.bulk.eset
```

#### Single cell data

The single cell data are from [ArrayExpression (E-MTAB-5060)](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-5061/), which contrains read counts for 25453 genes of 2209 cells. The read counts are available on the [data download page](), in the form of an `ExpressionSet`:

``` r
EMTAB.eset = readRDS('https://github.com/xuranw/MuSiC/tree/master/vignettes/data/EMTABesethealthy.rds')
EMTAB.eset
```

The cell types are identified by tSNE clustering.

### Estimation of cell type proportions

We build a reference basis matrix and cross-sample variance from the single cell gene expression, which we use to infer the composition of bulk tisue samples from there total gene expression.

#### Bulkding the reference basis matrices

The purpose of this matrix is to provide reference expression profiles that t
