---
title: "MuSiC: Sample Analysis"
output: html_document
---

Examples
--------

We reproduce here the analysis detailed in MuSiC manuscript: 

- Total gene expression data were obtained from GEO dataset [GSE50244](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50244), which contains read counts data on gobal transcriptomic analysis of bulk human pancreatic islets to study glucose metabolism in healthy and hyper-glycemic conditions (Fadista et al. 2014).

-   Single cell RNA-seq data were generated by ArrayExpress dataset [E-MTAB-5061](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-5061/), which contains read counts data on single cell mRNA-seq of human pancreatic islets from 10 donors, 6 healthy donors and 4 Type II diabetes donors (Segerstolpe et al. 2016). We used single cell data from 6 healthy donors as reference in deconvolution.

### Data Preparation

Bioconduct base package provides `ExpressionSet` class, which is a convenient data structure to hold expression data along with sample/feature annotation. Here we use two `ExpressionSet` objects to handle the bulk and single cell data respectively.

#### Bulk data

The dataset's [GEO entry (GSE50244)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50244) contains raw RNA-seq and sample annotation data. For the purpose of this vignette, we will use the read counts data on the [this page](https://github.com/xuranw/MuSiC/tree/master/vignettes/data/GSE50244bulkeset.rds). Please download this file.

``` r
# Download GSE50244 dataset from Github
GSE50244.bulk.eset = readRDS('yourpath/GSE50244bulkeset.rds')
GSE50244.bulk.eset
```

#### Single cell data

The single cell data are from [ArrayExpression (E-MTAB-5061)](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-5061/), which contrains read counts for 25453 genes of 2209 cells. The read counts are available on the [this page](https://github.com/xuranw/MuSiC/tree/master/vignettes/data/EMTABesethealthy.rds), in the form of an `ExpressionSet`. Please download this file.

``` r
# Download EMTAB single cell dataset from Github
EMTAB.eset = readRDS('yourpath/EMTABesethealthy.rds')
EMTAB.eset
```

The cell types are identified by tSNE clustering by Segerstolpe et al.

### Estimation of cell type proportions

In stead of selecting marker genes, MuSiC gives weights to each gene. The weight scheme is based on cross-subject variation: up-weight gene with low variation and down-weight gene with high variation. The composition of bulk tisue samples from there total gene expression is inferred by both cross-subject mean and variance by all genes.

The deconvolution of 89 subjects from Fadista et al. can be realized by bulk data `GSE50244.bulk.eset` and single cell data `EMTAB.eset`. We constrained our estimation on 6 major cell type: alpha, beta, delta, gamma, acinar and ductal, which take up over 90% of the whole islet. Function `music_prop` is the function for estimation. We can specify the way of grouping scRNA-seq data by `cluster`. `samples` refers to different subjects and `select.ct` refers to cell type included. The estimated proportions are normalized based on included cell type.

``` r
# Estimation cell type proportions
Est.prop.GSE50244 = music_prop(bulk.eset = GSE50244.bulk.eset, sc.eset = EMTAB.eset, clusters = 'cellType', samples = 'sampleID', select.ct = c('alpha', 'beta', 'delta', 'gamma', 'acinar', 'ductal'))

Prop.GSE50244.MuSiC = Est.prop.GSE50244$Est.prop.weighted
```
`music_prop` not only provides MuSiC estimated cell type proportions (`Est.prop.weighted`), but also Non-negative least square estimated proportions (`Est.prop.allgene`). It also returns the weights (`Weight.gene`).